persistence/sql/persister_consent.go

```diff
@@ -517,6 +518,32 @@ func (p *Persister) FlushInactiveLoginConsentRequests(ctx context.Context, notAf
 			time.Now().Add(-p.config.ConsentRequestMaxAge()),
 			notAfter).Exec()
 
+		if err != nil {
+			return sqlcon.HandleError(err)
+		}
+
+		err = p.Connection(ctx).RawQuery(fmt.Sprintf(`
+			DELETE
+			FROM %[1]s
+			WHERE NOT EXISTS
+				(
+				SELECT NULL
+				FROM %[2]s
+				WHERE %[2]s.login_session_id = %[1]s.id
+				)
+			AND NOT EXISTS
+				(
+				SELECT NULL
+				FROM %[3]s
+				WHERE %[3]s.login_session_id = %[1]s.id
+				)
+			AND authenticated_at < ?
+			`,
+			(&ls).TableName(),
+			(&lr).TableName(),
+			(&cr).TableName()),
+			notAfter).Exec()
```

Comment:
I think this is problematic because the `notAfter` has had a different context before. Now, this could remove sessions where users are signed in and that are not expired. I think not having any references is already a good indicator if this should be removed or not but I think we probably need to move this into it's own function and then add here

https://github.com/ory/hydra/blob/be8de37caaf8060cc7e56ccc45ab7d4df7a75d28/cmd/cli/handler_janitor.go#L131-L136

as an additional case.

Reply:
@aeneasr You are right! The `NOT EXISTS` conditions should be sufficient. The `notAfter` condition could be an extra peace of mind not to delete unintended records. It also could potentially help with the performance of the query since the engine may require to do fewer row scans this way (I didn't do any sort of benchmarking, I am just guessing).

I can remove it if you think that's misleading.